modules/tinymce/src/core/main/ts/api/util/Tools.ts

```diff
@@ -82,24 +83,17 @@ const is = (obj: any, type: string) => {
  * @param {Object} map Optional map to add items to.
  * @return {Object} Name/value map of items.
  */
-const makeMap = (items, delim?, map?) => {
-  let i;
+const makeMap = (items: ArrayLike<string> | string | undefined, delim?: string | RegExp, map?: Record<string, {}>): Record<string, {}> => {
+  const resolvedItems = Type.isString(items) ? items.split(delim || ',') : (items || []);
 
-  items = items || [];
-  delim = delim || ',';
+  const newMap: Record<string, {}> = map ? { ...map } : {};
 
-  if (typeof items === 'string') {
-    items = items.split(delim);
-  }
-
-  map = map || {};
-
-  i = items.length;
+  let i = resolvedItems.length;
   while (i--) {
-    map[items[i]] = {};
+    newMap[resolvedItems[i]] = {};
   }
 
-  return map;
+  return newMap;
 };
```

Comment:
This is a rework to make it work nicer with types. It also prevents mutating in the passed `map` variable.

Reply:
Hmm I understood that a different way, to me that meant that it'll extend it. I can see your point though so yeah I'll revert that part so it keeps mutating it.

Reply:
Worth noting there's very few places this is used in TinyMCE... the only place I can recall is in the schema where it chains multiple `mapMap` functions like so: 
```
const value = makeMap(defaultValue, ' ', makeMap(defaultValue.toUpperCase(), ' '));
```

Reply:
Have restored the mutation.